src/odemis/driver/xt_client.py

```diff
@@ -22,37 +21,72 @@
 """
 from __future__ import division, print_function
 
+import base64
 import logging
+import threading
+import time
+from concurrent.futures import CancelledError
 
-import msgpack_numpy as mn
-import zerorpc
+import msgpack
+import msgpack_numpy as m
+from Pyro5.api import Proxy
 
-mn.patch()
+from odemis import model
+from odemis import util
+from odemis.model import CancellableThreadPoolExecutor, HwError, isasync, CancellableFuture
 
-# Necessary, otherwise nothing at all will be shown
-logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)-15s: %(message)s")
-logging.getLogger().setLevel(logging.INFO)
+XT_RUN = "run"
+XT_STOP = "stop"
+XT_CANCEL = "cancel"
 
 
-class MicroscopeClient:
+class SEM(model.HwComponent):
     """
     Class to communicate with a Microscope server via the ZeroRPC protocol.
     """
 
-    def __init__(self, server_address="tcp://192.168.1.1:4242", timeout=30):
+    def __init__(self, name, role, children, address, timeout, daemon=None,
```

Comment:
Please provide a default value for the timeout (30s?), as it'll hint on what is a good value. I think we could even argue that the user shouldn't need to set it, and it's up to you to get a good constant.